### ISSUE TITLE
Maccy to mask entry

### ISSUE THREAD

**Issue by moneyease** [p0deje/Maccy#157]
I know I am asking you too much, but it'll be great if we can obfuscate some characters in entries while showing panel. Sometimes we are sharing screen and it'll be handy.

**Comment by p0deje** [comment]
I'm not sure I follow how that is supposed to work, can you provide more details? Maybe show an example of unobfuscated vs obfuscated entry.

**Comment by moneyease** [comment]
Thanks Alex,

I just created a mockup, hope it helps. I would rephrase like the user can
mask any entry of choice.

Manish


[image: Untitled.png]

**Comment by p0deje** [comment]
@moneyease I can't see any image, can you re-upload?

**Comment by moneyease** [comment]
I am having difficulty uploading to GitHub. 

Here is a link. 

https://pasteboard.co/JotHSM6.png

Manish

**Comment by p0deje** [comment]
Ok, I got your idea. Still, I don't understand how the masking should work. Would you mask some of the items one by one? Would you enable "masking mode" that would start masking all the items?

**Comment by moneyease** [comment]
How about just mark one by one (like we pin items).

**Comment by p0deje** [comment]
How valuable that would be? Are you going to mask mulitple items one-by-one before starting screen share? Maybe it might be easier to simply remove these items?

**Comment by moneyease** [comment]
it's like, I keep all sensitive entries masked all the time. If I happen to use my screen etc, I don't have to worry about making it right before sharing.

For example:

- Like I just copied a password, I go to Maccy and mask it forever using hotkey (TBD). ["passworrd" is masked as "pa****rd"]
- Anytime I need it, I just use Maccy to paste it for me (masked entries are still copied correctly). ["pa****rd" is still copied as "password".]
- If I need to delete, just delete it like a regular entry.
- If I need to unmask, just press hot-key (TBD). ["pa****rd" is change to "password".]

**Comment by p0deje** [comment]
Should the search still find symbols that are masked? E.g. if I have "pa****rd" and I search for "pass" - should it be found or not?

**Comment by moneyease** [comment]
It'll be good to have but not a must-have.

If you treat '*' as a wildcard in pa****rd, then pattern 'pass' should
match.

**Comment by p0deje** [comment]
1. Should "pa****rd" (masked "password") be found when I search for "pa11"? If not, it should be easier to implement.
2. Do we need to mask every symbol (e.g. 4 * to mask "sswo" in "password") or we can just use a fixed-length mask regardless of number of masked symbols (e.g. *** for "sswo" and "anyrandomnummberofsymbols")?

**Comment by moneyease** [comment]
Please see inline

On Mon, Sep 14, 2020 at 10:40 AM Alex Rodionov <notifications@github.com>
wrote:

>
>    1. Should "pa****rd" (masked "password") be found when I search for
>    "pa11"? If not, it should be easier to implement.
>
> Actually that would be a security concern as someone can guess your
password, although not a problem here.

>
>    1. Do we need to mask every symbol (e.g. 4 * to mask "sswo" in
>    "password") or we can just use a fixed-length mask regardless of number of
>    masked symbols (e.g. *** for "sswo" and "anyrandomnummberofsymbols")?
>
> I think we can just mask half of characters,
-> p -> p
-> pa -> p*
-> pas -> p*s
->pass -> p**S
-> passw -> pa**w
-> passwo -> pa***o
and so on ...

**Comment by p0deje** [comment]
There is now a way to rename pinned items, so I guess it should cover the described use case.
1. Copy password.
2. Pin it
3. Rename to something that makes no sense.
4. Use it for password pasting.
